plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -270,9 +270,22 @@ object ScalafmtPlugin extends AutoPlugin {
   private lazy val metabuildSources = Def.task {
     val rootBase = (LocalRootProject / baseDirectory).value
     val thisBase = (thisProject.value).base
-    if (rootBase == thisBase)
-      (BuildPaths.projectStandard(thisBase) ** GlobFilter("*.scala")).get
-    else Nil
+
+    if (rootBase == thisBase) {
+      val projectDirectory = BuildPaths.projectStandard(thisBase)
+      val targetDirectory =
+        BuildPaths.outputDirectory(projectDirectory).getAbsolutePath
+      println(projectDirectory)
+      projectDirectory
+        .descendantsExcept(
+          "*.scala",
+          (pathname: File) =>
+            pathname.getAbsolutePath.startsWith(targetDirectory)
```

Comment:
I guess this will not exclude target directories from meta-meta builds?

Reply:
I think that `project.excludeFilters` would be enough for this